### PR TITLE
[REG-1986] Create setup & installation guide

### PR DESCRIPTION
**Pull request by abeizer**
- Filled in the "Installing Regression Games" page.
- Moved "API Keys" page to "Core Concepts" instead of leaving it under "Getting Started"
- Removed duplicate info from the "creating your first automated test" section in favor of saying "go follow this setup guide for your preferred sample project".

https://www.loom.com/share/8ba78de31fac4f1dbf36787a552453e1

---

Find the pull request instructions [here](https://www.notion.so/regressiongg/Pull-Request-Process-0d57a7eb582a446983edfacafa406f1e?pvs=4)

Every reviewer and the owner of the PR should consider these points in their request (feel free to copy this checklist so you can fill it out yourself in the overall PR comment)

- [ ] The code is extensible and backward compatible
- [ ] New public interfaces are extensible and open to backward compatibility in the future
- [ ] If preparing to remove a field in the future (i.e. this PR removes an argument), the argument stays but is no longer functional, and attaches a deprecation warning. A linear task is also created to track this deletion task.
- [ ] Non-critical or potentially modifiable arguments are optional
- [ ] Breaking changes and the approach to handling them have been verified with the team (in the Linear task, design doc, or PR itself)
- [ ] The code is easy to read
- [ ] Unit tests are added for expected and edge cases
- [ ] Integration tests are added for expected and edge cases
- [ ] Functions and classes are documented
- [ ] Migrations for both up and down operations are completed
- [ ] A documentation PR is created and being reviewed for anything in this PR that requires knowledge to use
- [ ] Implications on other dependent code (i.e. sample games and sample bots) is considered, mentioned, and properly handled
- [ ] Style changes and other non-blocking changes are marked as non-blocking from reviewers

### CODE REVIEW

File: docs/getting-started/installing-regression-games.mdx [hunk 1]
```diff
@@ -4,6 +4,110 @@ sidebar_label: 'Installing Regression Games'
...
+
+Our core package provides an in-game overlay that acts as a UI for features such as performing
+[input recording and playback](../core-concepts/gameplay-sessions/gameplay-sessions-getting-started)
+and constructing and running [Bot Sequences](../core-concepts/bot-sequences/getting-started-with-bot-sequences-and-segments),
+but it also drives our data collection behind-the-scenes.
+This means the overlay prefab **must** be present in each scene where you want to use any Regression Games features.
+However, you can hide the visibility of the overlay through the project settings under
+`Edit` > `Project Settings` > `Regression Games` if needed.
+
+Find the `RGOverlayCanvasV2` prefab by navigating to
```
Comment (addisonbgross) [inline docs/getting-started/installing-regression-games.mdx:95]: We'll want to use the `RGOverlayCanvas` without the `V2`, it has the new Sequences stuff

**Review by addisonbgross — APPROVED** [review]
Just 1 small change, otherwise looks good

File: docs/getting-started/installing-regression-games.mdx (hunk 1 above)
Comment (abeizer) [reply docs/getting-started/installing-regression-games.mdx]: Good call. Since we're using an mvp branch to avoid releasing these docs until everything is ready I'll make sure to capture info for the upcoming changes rather than what we have public now

File: docs/changelog.mdx [hunk 2]
```diff
@@ -312,7 +312,7 @@ us at info@regression.gg for dedicated support and access.
 
 - [Agent Builder](404) Beta - Our GPT-enabled agent builder product is live for select players, now with improved validation tools. If you&apos;d like to try it out, let us know!
 - Our documentation site is now live! This docs site includes API/SDK references, tutorials, guides, and FAQs for both players and developers.
-- [API Keys](getting-started/api-keys/authenticating-with-api-keys) - developers and players can now create API keys to interact with RG services
+- [API Keys](core-concepts/authenticating-with-api-keys) - developers and players can now create API keys to interact with RG services
```
Comment (RG-nAmKcAz) [inline docs/changelog.mdx:315]: getting-started/installing-regression-games already has a section about the api-keys in it.

I don't want to have to maintain that content in 2 places.  Can we either inline this new content there or do something so that we don't have 2 separate writeups on similar content

**Review by RG-nAmKcAz — CHANGES_REQUESTED** [review]
Just the 1 comment about api keys docs and how they might fit into the install section without having 2 separate sections for it

File: docs/core-concepts/authenticating-with-api-keys.mdx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+sidebar_label: 'Authenticating with API Keys'
+---
+
+# Authenticating with API Keys
+
+The Regression Games platform provides the ability to generate API keys, which can be used
+to authenticate against our API. This is the recommended way to authenticate against our API
+when connecting external systems to Regression Games.
+
+You can find the **API Keys** manager at the bottom of your [Regression Games account settings](https://play.regression.gg/account).
```
Comment (batu) [inline docs/core-concepts/authenticating-with-api-keys.mdx:14]: When I clicked on the link my brain looked for API Keys Manager, maybe calling it API Keys section will be better.

File: docs/core-concepts/authenticating-with-api-keys.mdx [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+## Generating an API Key
+
+Enter a unique name for your key and then click **Generate Key**.
+API keys have a default expiration date of 1 year.
```
Comment (batu) [inline docs/core-concepts/authenticating-with-api-keys.mdx]: Can we change the default? If not we can just say 'API keys have an expiration date of 1 year.'

File: docs/core-concepts/authenticating-with-api-keys.mdx [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+![Create an API Key](img/api-keys/create-new-api-key.gif)
+
+
+## Using the API Key
+
+Once you have generated an API key, you can use it to authenticate against our API.
+
+### API Requests
+
+Include the key in the `Authorization` header of your request.
```
Comment (batu) [inline docs/core-concepts/authenticating-with-api-keys.mdx:26]: I believe in the flow we are pushing the user will:
> Go to Installing Regression Games page
> Get to Set an API Key
> Click on 'Learn more about our API keys, see the Authentica ...` 
> Come here.

Which means that this information doesn't have a lot of context, how do I send a request? What is a request? Consider moving this down and/or adding a link to the relevant page explaining the request.

File: docs/getting-started/installing-regression-games.mdx [hunk 6]
```diff
@@ -4,6 +4,115 @@ sidebar_label: 'Installing Regression Games'
...
+Open your Unity project and navigate to `Window` > `Package Manager`. From here, find the `Add package from git URL` and paste the following URL:
+
+```
+https://github.com/Regression-Games/RGUnityBots.git?path=src/gg.regression.unity.bots#v0.0.27
+```
+
+:::info
+- This package utilizes TextMeshPro. If you are prompted by Unity to add TextMeshPro assets to your project, please add them.
+- This package installs the [Newtonsoft Json](https://docs.unity3d.com/Packages/com.unity.nuget.newtonsoft-json@3.1/manual/index.html) dependency for serializing/deserializing Json message payloads.
+- If you get a note about Unity's new input system, click **Yes** and allow the editor to restart.
```
Comment (batu) [inline docs/getting-started/installing-regression-games.mdx:24]: Reading this line as a developer, without seeing the actual pop-up, can make me a bit concerned. I might think "Is adding regression sdk gonna change my whole input set up!?" consider adding a sentence to placate that potential fear/ add little bit more context.

If the pop-up itself is self explanatory this might be less of an issue, but we can't assume folks will read docs while going along.

File: docs/getting-started/installing-regression-games.mdx [hunk 7]
```diff
@@ -4,6 +4,115 @@ sidebar_label: 'Installing Regression Games'
...
+![Add the Regression Games core SDK to your Unity project](img/installing-regression-games/installing-sdk.gif)
+
+## Add Extension Packages to Your Project
+
+We provide the following extensions for our core SDK to support common project dependencies.
+Each of these can be added to your project using a git url, similar to core package installation.
+
+### Animation Rigging Extension
+
+Adds support for Unity Animation Rigging.
```
Comment (batu) [inline docs/getting-started/installing-regression-games.mdx:36]: Maybe add a sentence about how we are using this? When, as a user, should I add this? We can potentially kick this to the docs of the part that uses Animation Rigging. Given that this is the first page, the more streamlined we can be the better.

File: docs/getting-started/installing-regression-games.mdx [hunk 8]
```diff
@@ -4,6 +4,115 @@ sidebar_label: 'Installing Regression Games'
...
+
+```
+https://github.com/Regression-Games/RGUnityBots?path=src/gg.regression.unity.bots.animation.rigging#v0.0.27
+```
+
+### Entity Component System (ECS) Extension
+
+Enables game state collection of Unity ECS Entities and their Components.
+Without this extension, the core SDK can still capture GameObject data in projects using ECS but will have no access to ECS Entities.
+Depending on your project and its needs, this extension may not be necessary, but we highly recommend including it
```
Comment (batu) [inline docs/getting-started/installing-regression-games.mdx:46]: Do recommend it for projects that are not using ECS as well? Reading this that was my question.

File: docs/getting-started/installing-regression-games.mdx [hunk 9]
```diff
@@ -4,6 +4,115 @@ sidebar_label: 'Installing Regression Games'
...
+for full visibility of your game's internal state.
+
+Capturing Entities and their Component data may negatively impact runtime performance.
```
Comment (batu) [inline docs/getting-started/installing-regression-games.mdx:49]: While this is important to communicate, I wonder whether this is the right spot for it? I don't yet know what will be captured, I am still installing things but now I might need to make a decision about performance.

File: docs/getting-started/installing-regression-games.mdx [hunk 10]
```diff
@@ -4,6 +4,115 @@ sidebar_label: 'Installing Regression Games'
...
+
+Generate a new API key through your account settings in the Regression Games client, and copy the generated value to your clipboard.
+
+![Generate a new API key](../core-concepts/img/api-keys/create-new-api-key.gif)
+
+:::info
+To learn more about our API keys, see the [Authenticating with API Keys](../core-concepts/authenticating-with-api-keys) section.
+:::
+
+Do not submit any information to the welcome prompt in Unity.
```
Comment (batu) [inline docs/getting-started/installing-regression-games.mdx]: If our docs are telling our user not to use our own welcome prompt, I think we should prioritize updating our welcome prompt and streamlining this process.

File: docs/getting-started/installing-regression-games.mdx [hunk 11]
```diff
@@ -4,6 +4,115 @@ sidebar_label: 'Installing Regression Games'
...
+and constructing and running [Bot Sequences](../core-concepts/bot-sequences/getting-started-with-bot-sequences-and-segments),
+but it also drives our data collection behind-the-scenes.
+This means the overlay prefab **must** be present in each scene where you want to use any Regression Games features.
+However, you can hide the visibility of the overlay through the project settings under
+`Edit` > `Project Settings` > `Regression Games` if needed.
+
+Find the `RGOverlayCanvas` prefab by navigating to
+`Packages` > `Regression Games Unity Bots` > `Runtime` > `Prefabs` and dragging it into your scene's object hierarchy.
+Make sure that the canvas is at the top of the hierarchy, so that it is always visible and interactable.
+Repeat this for each scene that must have access to Regression Games features.
```
Comment (batu) [inline docs/getting-started/installing-regression-games.mdx:90]: Can we get away with adding it once and setting it 'don't destroy on load'?

File: docs/getting-started/installing-regression-games.mdx [hunk 12]
```diff
@@ -4,6 +4,115 @@ sidebar_label: 'Installing Regression Games'
...
+You can add one by right-clicking in your scene's Hierarchy and selecting `GameObject` > `UI` > `Event System`.
+If you are using any of the sample scenes we provide in our guides, this is already included.
+:::
+
+
+## Next Steps
+
+The Regression Games SDK is now set up and ready to be explored.
+Visit our documentation on the [Core Concepts](../core-concepts/authenticating-with-api-keys) that drive features in our SDK and client,
+or head over to the [Guides](../guides/writing-advanced-validations) section for walkthroughs and examples of real-world applications.
```
Comment (batu) [inline docs/getting-started/installing-regression-games.mdx:105]: I think at somepoint we should be opinionated in here and tell the users which is the best next step is. (in addition to giving them these options). Given that the next page is Creating your first automated test, that might be added here explicitly.

File: docs/getting-started/installing-regression-games.mdx [hunk 13]
```diff
@@ -4,6 +4,115 @@ sidebar_label: 'Installing Regression Games'
...
+Generate a new API key through your account settings in the Regression Games client, and copy the generated value to your clipboard.
+
+![Generate a new API key](../core-concepts/img/api-keys/create-new-api-key.gif)
+
+:::info
+To learn more about our API keys, see the [Authenticating with API Keys](../core-concepts/authenticating-with-api-keys) section.
+:::
+
+Do not submit any information to the welcome prompt in Unity.
+Instead, close the prompt and navigate to `Edit` > `Project Settings` > `Regression Games`.
```
Comment (vontell) [inline docs/getting-started/installing-regression-games.mdx]: Can we remove this? Or rather, this should be updated to reflect the changes in the new PR

File: docs/changelog.mdx (hunk 2 above)
Comment (abeizer) [reply docs/changelog.mdx:315]: Created two partials for "create key" and "set key in unity". This is what the relevant section of the installation guide looks like now

image

**Review by RG-nAmKcAz — APPROVED** [review]

File: docs/getting-started/creating-your-first-automated-test.mdx [hunk 14]
```diff
@@ -0,0 +1,132 @@
+---
+sidebar_label: 'Creating Your First Automated Test'
+---
+
+# Creating Your First Automated Test
```
Comment (batu) [inline docs/getting-started/creating-your-first-automated-test.mdx:5]: I think this is not new text, so will hold of on giving feedback on this

**Review by batu — APPROVED** [review]
Looks good overall! Added a bunch of notes, take it or leave it as you find them useful

File: sidebars.js [hunk 15]
```diff
@@ -39,6 +38,7 @@ const sidebars = {
       type: "category",
       label: "Core Concepts",
       items: [
+        "core-concepts/authenticating-with-api-keys",
```
Comment (vontell) [inline sidebars.js:41]: Does this mean that the API key is the first thing in the core concepts? I'm not too big of a fan of API keys showing up as a "Core Concept" at all though, to me core concepts are things about how we are solving their problems in automated QA, and I don't really consider the ability to authenticate as a core feature they would care about in that context. Maybe this should be in its own Account Management section?

**Review by vontell — CHANGES_REQUESTED** [review]

File: docs/getting-started/installing-regression-games.mdx (hunk 12 above)
Comment (vontell) [reply docs/getting-started/installing-regression-games.mdx:105]: Once some more guides are made (waiting on me), I can go and link a few here :)

File: docs/getting-started/installing-regression-games.mdx (hunk 6 above)
Comment (abeizer) [reply docs/getting-started/installing-regression-games.mdx:24]: I don't actually know what this pop-up looks like. @RG-nAmKcAz I believe you've seen the pop ups for the "new input system" - do you have thoughts?

File: docs/getting-started/installing-regression-games.mdx (hunk 7 above)
Comment (abeizer) [reply docs/getting-started/installing-regression-games.mdx:36]: I actually don't know what happens if you don't have this extension in projects using Animation Rigging. @RG-nAmKcAz can you advise on what sort of data this captures / what the effects are of not including it?

File: docs/core-concepts/authenticating-with-api-keys.mdx (hunk 3 above)
Comment (abeizer) [reply docs/core-concepts/authenticating-with-api-keys.mdx]: changed to "section"

File: docs/core-concepts/authenticating-with-api-keys.mdx (hunk 4 above)
Comment (abeizer) [reply docs/core-concepts/authenticating-with-api-keys.mdx]: We cannot change the default from the client today. Removed the "default" bit

File: docs/getting-started/installing-regression-games.mdx (hunk 7 above)
Comment (RG-nAmKcAz) [reply docs/getting-started/installing-regression-games.mdx:36]: you get state data for animation rigging components in the state whereas you will not without it

File: docs/getting-started/installing-regression-games.mdx (hunk 9 above)
Comment (abeizer) [reply docs/getting-started/installing-regression-games.mdx:51]: We don't have a separate section for extension projects so right now there's nowhere reasonable to move this disclaimed to. If we end up with a bunch of different extensions then it'd make sense to move them to their own page, but for now I don't want to divert the user from this guide if there isn't enough content to justify it.

I do include that the user can toggle off these features so they don't need to make any permanent decision about performance. It's recommended but ultimately optional and can be added at any time (maybe I'll include that last part too)

File: docs/getting-started/installing-regression-games.mdx (hunk 10 above)
Comment (abeizer) [reply docs/getting-started/installing-regression-games.mdx]: This was removed while you were reviewing. We've decided to remove email login as an option altogether, so I've gone ahead and assumed that email/password login won't exist once these docs are live.

File: docs/getting-started/installing-regression-games.mdx (hunk 6 above)
Comment (RG-nAmKcAz) [reply docs/getting-started/installing-regression-games.mdx:24]: Personally as a Unity dev, I find that the popup is self explanatory.  It tells you to restart to enable the new input system reading native device states.

File: docs/getting-started/installing-regression-games.mdx (hunk 12 above)
Comment (abeizer) [reply docs/getting-started/installing-regression-games.mdx]: Good point. This now directs the user to the creating-your-first-automated-test page and our guides sections explicitly

File: docs/getting-started/creating-your-first-automated-test.mdx (hunk 14 above)
Comment (abeizer) [reply docs/getting-started/creating-your-first-automated-test.mdx:5]: Yeah this was renamed, moved from its previous url, and I updated some wording here and there but the content itself is old

File: docs/getting-started/installing-regression-games.mdx (hunk 11 above)
Comment (RG-nAmKcAz) [reply docs/getting-started/installing-regression-games.mdx:93]: We can either leave this as is, or write a paragraph explaining to them that it needs to be in your 'first' scene in all situations.  Telling them every scene was 'safer' as it is not uncommon to use one scene as the first scene in the IDE but a different one in a built runtime.

File: sidebars.js (hunk 15 above)
Comment (abeizer) [reply sidebars.js:41]: Why don't we see how the other pages flesh out and then we can move things around more if this doesn't make sense to keep here? Updating links to pages and images has been time-consuming and I'm more concerned with getting the content _out there_ for people to use

File: docs/getting-started/installing-regression-games.mdx (hunk 6 above)
Comment (vontell) [reply docs/getting-started/installing-regression-games.mdx:24]: This pops up for me all the time since I make a lot of fresh projects from Boss Room. The dialog does have a good amount of information on what will happen, but yes I think adding a quick comment would be good. I personally don't know how much this changes in the project and how easy it is to "undo" the operation. Maybe we can just put something like 

```suggestion
- If you get a note about Unity's new input system, click **Yes** and allow the editor to restart (this will allow the new input system to be enabled).
```

Here is what it looks like: 
image

File: docs/getting-started/installing-regression-games.mdx (hunk 13 above)
Comment (vontell) [reply docs/getting-started/installing-regression-games.mdx]: Thanks!

File: docs/core-concepts/authenticating-with-api-keys.mdx (hunk 5 above)
Comment (abeizer) [reply docs/core-concepts/authenticating-with-api-keys.mdx:26]: I can think of two options:
1. Remove the link to this page from the installation guide and assume that if they want to know more then they'll find the api-keys doc themselves. I don't like this because we don't have any search functionality on the site and security is something Zack and Aaron have mentioned as factors that could scare off potential users if not handled correctly.
2. Move the "request" section below the "unity" section so that the order is more relevant to users coming to this page from the installation guide.

> and/or adding a link to the relevant page explaining the request.
This part I disagree with. If a developer doesn't already know what an API request is or how headers work, then this section either isn't relevant to them or they need to learn more about REST APIs in general and it's not our job to explain teach that to them.

File: docs/getting-started/installing-regression-games.mdx (hunk 7 above)
Comment (abeizer) [reply docs/getting-started/installing-regression-games.mdx:36]: That's what I figured, wanted to make sure it wouldn't break anything if a project uses animation rigging but doesn't include this extension

File: docs/getting-started/installing-regression-games.mdx (hunk 11 above)
Comment (abeizer) [reply docs/getting-started/installing-regression-games.mdx:95]: I'm leaning towards the "safer" route. If we receive negative feedback on it then we can clarify

**Review by vontell — APPROVED** [review]

File: docs/core-concepts/authenticating-with-api-keys.mdx (hunk 5 above)
Comment (batu) [reply docs/core-concepts/authenticating-with-api-keys.mdx]: The point isn't necessarily to teach people about the REST API's in general, but how requests are used in the context of our tool. When we have a page telling people about that we can link it there
